src/main/kotlin/com/sourcegraph/cody/agent/protocol/TextEdit.kt

```diff
@@ -0,0 +1,15 @@
+package com.sourcegraph.cody.agent.protocol
+
+data class TextEdit(
+    // This tag will be 'replace', 'insert', or 'delete'.
+    val type: String,
+
+    // Valid for replace & delete.
+    val range: Range? = null,
```

Comment:
Wouldn't it make sense to create two different types of edits, one position based and one renge based? This way we don't need this comments and all fields would always be required and used.

Reply:
Probably, but it's a bigger change; maybe we can wait until after this PR is merged?